keps/sig-storage/20190125-online-growing-persistent-volume-size.md

```diff
@@ -81,187 +81,25 @@ Enable users to increase the size of a PVC which is already in use (mounted). Th
-
-- Since we support only `xfs` and `ext3/4`, we needn't worry about exotic file systems that can be attached/mounted to multiple nodes at the same time and require a resizing operation on a node, such as [gfs2](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/global_file_system_2/s1-manage-growfs).
-
-##### File System Resizing Operation
-
-Currently `Reconciler` runs a periodic loop to reconcile the desired state of the world with the actual state of the world by triggering attach, detach, mount, and unmount operations. Based on this mechanism, we can make `Reconciler` trigger file system resize operations too.
+When adding a volume to `DesiredStateOfWorld`, the populator will include the PV & PVC fields needed to check whether the volume requires an online resizing operation: one is required if `PVC.Status.Capacity` is less than `PV.Spec.Capacity`.
 
-In each loop, for each volume in DSW, `Reconciler` checks whether it exists in ASW or not. If the volume exists and is marked as file system resizing required, ASW returns a `fsResizeRequiredError` error. Then `Reconciler` triggers a file system resizing operation for this volume. The code snippet looks like this:
+Later, `VolumeManager`'s `Reconciler` mounts the volume in `DesiredStateOfWorld` to a pod and adds it to `ActualStateOfWorld`. It will copy the `PVC.Status.Capacity` field from the `DesiredStateOfWorld` representation to the `ActualStateOfWorld` one.
 
-```go
-if cache.IsFSResizeRequiredError(err) {
-	glog.V(4).Infof(volumeToMount.GenerateMsgDetailed("Starting operationExecutor.VolumeFileSystemResize", ""))
-	err := rc.operationExecutor.VolumeFileSystemResize(
-		volumeToMount.VolumeToMount,
-		rc.actualStateOfWorld)
-	if err != nil &&
-		!nestedpendingoperations.IsAlreadyExists(err) &&
-		!exponentialbackoff.IsExponentialBackoff(err) {
-		// Ignore nestedpendingoperations.IsAlreadyExists and exponentialbackoff.IsExponentialBackoff errors, they are expected.
-		// Log all other errors.
-		glog.Errorf(volumeToMount.GenerateErrorDetailed("operationExecutor.VolumeFileSystemResize failed", err).Error())
-	}
-	if err == nil {
-		glog.V(4).Infof(volumeToMount.GenerateMsgDetailed("operationExecutor.VolumeFileSystemResize started", ""))
-	}
-}
-```
+The reconciler will periodically check whether any volume that's mounted in a pod (as reported by `ActualStateOfWorld`) requires an online resizing operation by reading its fields and triggering an online file system resize operation if: the volume's `PVC.Status.Capacity` in `ActualStateOfWorld` is less than its `PV.Spec.Capacity` in `DesiredStateOfWorld`, the volume has `volumeMode` `Filesystem`, and the volume is not `readOnly`.
```

Comment:
Do we validate fstype anywhere?

Reply:
not until we are right about to calls resize2fs: https://github.com/kubernetes/kubernetes/blob/a3ccea9d8743f2ff82e41b6c2af6dc2c41dc7b10/pkg/util/resizefs/resizefs_linux.go#L54

is fstype guaranteed to be filled? if so we can check it.

Reply:
I think it can be empty